cli/cmd/generate.go

```diff
@@ -0,0 +1,66 @@
+package cmd
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/sirupsen/logrus"
+
+	generator "github.com/opencontrol/oscalkit/generator"
+	"github.com/opencontrol/oscalkit/templates"
+	"github.com/opencontrol/oscalkit/types/oscal/catalog"
+	"github.com/urfave/cli"
+)
+
+var profilePath string
+
+//Generate Cli command to generate go code for controls
+var Generate = cli.Command{
+	Name:  "generate",
+	Usage: "generates go code against provided catalogs and profile",
+	Flags: []cli.Flag{
+		cli.StringFlag{
+			Name:        "profile, p",
+			Usage:       "profile to intersect against",
+			Destination: &profilePath,
+		},
+	},
+	Before: func(c *cli.Context) error {
+		if profilePath == "" {
+			return cli.NewExitError("oscalkit sign is missing the --profile flag", 1)
+		}
+
+		return nil
+	},
+	Action: func(c *cli.Context) error {
+
+		bytes, err := ioutil.ReadFile(profilePath)
```

Comment:
I would instead get an `*os.File` type here instead of `[]byte` since `*os.File` is an `io.Reader` and can be passed directly into `oscal.New()`.  vs. using `bytes.NewReader()` in the `readOscal()` function.

You can use the `os.Stat()` function to determine whether or not the `profilePath` exsists too.

Reply:
Done